app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherAdapter.kt

```diff
@@ -292,9 +286,7 @@ class TabSwitcherAdapter(
 
     fun onTabMoved(from: Int, to: Int) {
         val swapped = list.swap(from, to)
-        list.clear()
-        list.addAll(swapped)
-        notifyItemMoved(from, to)
+        updateData(swapped)
```

Comment:
We can let the `DiffUtil` to notify the adapter.

Reply:
This is the fix. Because `move != swap`. Previous diff is only correct if swapping adjacent items, otherwise it would've needed:

```
val a = min(from, to)
val b = max(from, to)
notifyItemMoved(a, b)
if (b > a + 1) notifyItemMoved(b - 1, a)
```

But I think there's still a possible race condition from scheduling the `onMoveTab` handling:

PTAL https://github.com/duckduckgo/Android/pull/4881